pkg/innerring/processors/governance/process_update.go

```diff
@@ -103,3 +114,15 @@ func (gp *Processor) processAlphabetSync() {
 
 	gp.log.Info("finished alphabet list update")
 }
+
+func prettyKeys(keys keys.PublicKeys) string {
+	sb := strings.Builder{}
+	for _, key := range keys {
+		sb.WriteString(hex.EncodeToString(key.Bytes()))
+		sb.WriteString(",")
+	}
+
+	ln := sb.Len()
+
+	return sb.String()[:ln-1]
```

Comment:
We'll panic if `keys` is empty. I'd check this or add function comment `keys must not be empty`. 

Also double-check is the condition `if newAlphabet == nil` enough.

Reply:
Fixed panic.